caps/capabilities.go

```diff
@@ -33,12 +33,63 @@ type Capability struct {
 	// the behavior allowed and expected from providers and consumers of that
 	// capability, and also which information should be exchanged by these
 	// parties.
-	Type Type
+	Type *Type
 	// Attrs are key-value pairs that provide type-specific capability details.
 	Attrs map[string]string
 }
 
+var (
+	testCapability = &Capability{
+		Name:  "test-name",
+		Label: "test-label",
+		Type:  testType,
+		Attrs: nil,
+	}
+)
+
 // String representation of a capability.
 func (c Capability) String() string {
 	return c.Name
 }
+
+// CapabilityRepr is the JSON representation of a capability.
+// It exists so that Type can be replaced by Type.Name as we don't want to
+// create or describe capabilities fully, just to refer to them.
+type CapabilityRepr struct {
```

Comment:
No.

Repeating what I said on telegram:
CapabilityRepr is the wrong way to go about it: if you want to marshal caps.Type in a strange way, that's [probably] fine; just make caps.Type implement json.Marshaler and json.Unmarshaler, and the Repr thing just goes away.